Tasks/SqlDacpacDeploymentOnMachineGroup/ExecuteScriptsInTransaction.sql

```diff
@@ -0,0 +1,282 @@
+						CASE
+							WHEN CHARINDEX('-- Hash:', t.text) > 0 THEN REPLACE(SUBSTRING(t.Text, CHARINDEX('-- Hash:', t.text) + 51, 255), '@', ' ') 
+							ELSE REPLACE(SUBSTRING(t.text, 1, 255), '@', ' ')           
+						END +                                                           
+						'@stmt=' +
+						CASE WHEN a.statement_end_offset IS NOT NULL
+							THEN  REPLACE(SUBSTRING(t.text, 1 + a.statement_start_offset / 2, 255), '@', ' ')
+							ELSE NULL
+						END
+				FROM	sys.dm_tran_locks l
+				JOIN    sys.dm_exec_sessions  s WITH (NOLOCK)
+				ON      l.request_session_id = s.session_id
+				JOIN    sys.dm_exec_requests a WITH (NOLOCK)
+				ON      l.request_session_id = a.session_id
+				JOIN	sys.dm_exec_connections c WITH (NOLOCK)
+				ON		l.request_session_id = c.session_id
+				OUTER APPLY sys.dm_exec_sql_text(c.most_recent_sql_handle) t
+				WHERE	l.resource_database_id = DB_ID()
+				AND     l.request_status = 'GRANT'
+				AND		l.resource_type = 'APPLICATION'
+				AND     l.resource_description like '%[[]' + @_shortLockName + ']%'
+				AND     a.total_elapsed_time > @_longRunningThresholdMilliSeconds            
+				ORDER BY a.total_elapsed_time DESC
+			END
+
+			IF(@_statusMessage <> '')
+			BEGIN
+				RAISERROR('%s', 0, 231, @_statusMessage) WITH NOWAIT
+
+				SET @_statusMessage = 'Servicing lock attempt ' + CONVERT(VARCHAR(10), @_attempt) + ' result: Long running request detected.'
+				RAISERROR (@_statusMessage, 0, 231) WITH NOWAIT
+			END
+			ELSE 
+			BEGIN
+				SET @_acquireLockStart = GETUTCDATE()
+        
+				SET @_statusMessage = 'Acquiring servicing lock: @attempt=' + CONVERT(VARCHAR(10), @_attempt) + '@timestamp=' + CONVERT(VARCHAR, GETUTCDATE(), 109)
+				RAISERROR (@_statusMessage, 0, 231) WITH NOWAIT
+
+				if @_attempt = @_acquireLockMaxAttempts - @_lastAttemptsToKillSessions + 1
+					SET @_acquireLockMilliseconds = @_acquireLockMilliseconds + 10 * 1000
+
+				EXEC @_result = sp_getapplock  @Resource = @_acquireLock, @LockMode = 'Exclusive', @LockTimeout = @_acquireLockMilliseconds
+       
+				if @_result >= 0
+					BREAK
+
+				SET @_statusMessage = 'Servicing lock attempt ' + CONVERT(VARCHAR(10), @_attempt) + ' result: Timed out while acquiring the lock.'
+				RAISERROR (@_statusMessage, 0, 231) WITH NOWAIT
+			END
+
+			SET @_attempt = @_attempt + 1
+
+			SET @_noWaitAttempt = 1
+			WHILE @_noWaitAttempt <= 15
+			BEGIN
+				WAITFOR DELAY '00:00:01';
+				EXEC @_result = sp_getapplock  @Resource = @_acquireLock, @LockMode = 'Exclusive', @LockTimeout = 0
+				IF @_result >= 0
+					BREAK
+
+				SET @_noWaitAttempt = @_noWaitAttempt + 1
+			END
+
+			IF @_result >= 0
+				BREAK
+		END
+	END
+
+    IF @_result < 0
+    BEGIN
+        ROLLBACK
+		DROP VIEW _vw_tmpSqlFilesTable
+		DROP TABLE _tmpSqlFilesTable
+		PRINT 'Failed to acquire exclusive lock. Dropping temporary table, view and rolling back.'
+        RAISERROR('%%error="800070";%%:Failed to acquire an exclusive servicing lock', 16, -1)
```

Comment:
>800070 [](start = 28, length = 6)

I think, this is TFS specific. TFS converts this SQL error into C#  exception.
I am not able to see if we are doing the same in task